server/src/main/java/org/opensearch/indices/recovery/RecoverySettings.java

```diff
@@ -157,6 +157,17 @@ public class RecoverySettings {
         Property.NodeScope
     );
 
+    /**
+     * Controls minimum number of metadata files to keep in remote segment store.
+     * {@code value < 1} will disable deletion of stale segment metadata files.
+     */
+    public static final Setting<Integer> CLUSTER_REMOTE_INDEX_MIN_SEGMENT_METADATA_FILES_SETTING = Setting.intSetting(
+        "cluster.remote_store.index.min.segment-metadata",
+        10,
+        Property.NodeScope,
+        Property.Dynamic
+    );
```

Comment:
Lets restrict to a an upper bound to maybe 100 so that it doesn't impact storage size